File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
 public:
  void reverseString(vector<char>& s) {
    char temp = '\0';
    for (int i = 0; i < (s.size() / 2); i++) {
      temp = s[i];
      s[i] = s[s.size() - i - 1];
      s[s.size() - i - 1] = temp;
    }
  }
};

File: 0344-reverse-string/0344-reverse-string.py
# time complexity: O(n)
# space complexity: O(1)
from typing import List


class Solution:
    def reverseString(self, s: List[str]) -> None:
        for i in range(len(s) // 2):
            s[i], s[len(s) - i - 1] = s[len(s) - i - 1], s[i]
        return s


s = ["h", "e", "l", "l", "o"]
print(Solution().reverseString(s))
